packages/in_app_purchase/in_app_purchase_storekit/shared/Classes/FIAObjectTranslator.m

```diff
@@ -0,0 +1,297 @@
+  }
+  if (@available(iOS 11.2, *)) {
+    [map setObject:[FIAObjectTranslator getMapFromSKProductDiscount:product.introductoryPrice]
+                       ?: [NSNull null]
+            forKey:@"introductoryPrice"];
+  }
+  if (@available(iOS 12.2, *)) {
+    [map setObject:[FIAObjectTranslator getMapArrayFromSKProductDiscounts:product.discounts]
+            forKey:@"discounts"];
+  }
+  if (@available(iOS 12.0, *)) {
+    [map setObject:product.subscriptionGroupIdentifier ?: [NSNull null]
+            forKey:@"subscriptionGroupIdentifier"];
+  }
+  return map;
+}
+
++ (NSDictionary *)getMapFromSKProductSubscriptionPeriod:(SKProductSubscriptionPeriod *)period {
+  if (!period) {
+    return nil;
+  }
+  return @{@"numberOfUnits" : @(period.numberOfUnits), @"unit" : @(period.unit)};
+}
+
++ (nonnull NSArray *)getMapArrayFromSKProductDiscounts:
+    (nonnull NSArray<SKProductDiscount *> *)productDiscounts {
+  NSMutableArray *discountsMapArray = [[NSMutableArray alloc] init];
+
+  for (SKProductDiscount *productDiscount in productDiscounts) {
+    [discountsMapArray addObject:[FIAObjectTranslator getMapFromSKProductDiscount:productDiscount]];
+  }
+
+  return discountsMapArray;
+}
+
++ (NSDictionary *)getMapFromSKProductDiscount:(SKProductDiscount *)discount {
+  if (!discount) {
+    return nil;
+  }
+  NSMutableDictionary *map = [[NSMutableDictionary alloc] initWithDictionary:@{
+    @"price" : discount.price.description ?: [NSNull null],
+    @"numberOfPeriods" : @(discount.numberOfPeriods),
+    @"subscriptionPeriod" :
+            [FIAObjectTranslator getMapFromSKProductSubscriptionPeriod:discount.subscriptionPeriod]
+        ?: [NSNull null],
+    @"paymentMode" : @(discount.paymentMode),
+  }];
+  if (@available(iOS 12.2, *)) {
+    [map setObject:discount.identifier ?: [NSNull null] forKey:@"identifier"];
+    [map setObject:@(discount.type) forKey:@"type"];
+  }
+
+  // TODO(cyanglaz): NSLocale is a complex object, want to see the actual need of getting this
+  // expanded to a map. Matching android to only get the currencySymbol for now.
+  // https://github.com/flutter/flutter/issues/26610
+  [map setObject:[FIAObjectTranslator getMapFromNSLocale:discount.priceLocale] ?: [NSNull null]
+          forKey:@"priceLocale"];
+  return map;
+}
+
++ (NSDictionary *)getMapFromSKProductsResponse:(SKProductsResponse *)productResponse {
+  if (!productResponse) {
+    return nil;
+  }
+  NSMutableArray *productsMapArray = [NSMutableArray new];
+  for (SKProduct *product in productResponse.products) {
+    [productsMapArray addObject:[FIAObjectTranslator getMapFromSKProduct:product]];
+  }
+  return @{
+    @"products" : productsMapArray,
+    @"invalidProductIdentifiers" : productResponse.invalidProductIdentifiers ?: @[]
+  };
+}
+
++ (NSDictionary *)getMapFromSKPayment:(SKPayment *)payment {
+  if (!payment) {
+    return nil;
+  }
+  NSMutableDictionary *map = [[NSMutableDictionary alloc] initWithDictionary:@{
```

Comment:
There are many magic strings below. Could you leave a comment with a link that explains these values?

Reply:
Just to be sure, [this comment](https://github.com/flutter/plugins/pull/6517#discussion_r1029704543) also applies here, right?
> This is just moved code; the scope of the review should be to enabling macOS support within the existing code, rather than changing/improving the code more generally.

Reply:
Yes, I followed up with @a-wallen offline about reviewing again without including b4e810f5493f7fe2732771802568cb335441b8ab.

Reply:
Thanks for the clarification.